staging_vespalib/src/vespa/vespalib/metrics/name_repo.h

```diff
@@ -0,0 +1,47 @@
+// Copyright 2018 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+#pragma once
+
+#include <vespa/vespalib/stllike/string.h>
+#include "dimension.h"
+#include "label.h"
+#include "metric_name.h"
+#include "point.h"
+
+#include "name_collection.h"
+#include "point_map_collection.h"
+
+namespace vespalib {
+namespace metrics {
+
+/**
+ * Simple repo class
+ **/
+class NameRepo
+{
+private:
+    NameCollection _metricNames;
```

Comment:
in the future, when metrics and dimensions are orthogonal, consider moving the name into the actual metric